test.js

```diff
@@ -7,13 +7,89 @@ Travis CI runs all of the tests in the /tests directory when building.
 */
 
 var assert = require('assert');
+var myCode = require('./index');
 
-describe('LocationHandler', function() {
-    it('Demo test that should pass', function() {
-        assert.equal(-1, [1,2,3].indexOf(4));
+describe('Input validation tests', function() {
+    describe('UserID input validation tests', function() {
+      it('UserID test - testing a random 16 character ID', function() {
+        assert.equal(true, myCode.userIdRegExValid("abcdef1111111111"));
+      });
+      it('UserID test - testing a random 10 character ID', function() {
+        assert.equal(false, myCode.userIdRegExValid("abcd234517"));
+      });
+      //test fails---------? Fix Regex
+      it('UserID test - testing a random 17 character ID', function() {
```

Comment:
According to the Travis build, this test fails:

    AssertionError [ERR_ASSERTION]: false == true
      + expected - actual

This test should be amended (or the initial code in `index.js` should be fixed - @harrygcollins please can you investigate?) and then the Travis build should pass.

Reply:
From my RegEx experience, the {16} indicated a UserID with at most 16 characters. Given as input (for the test) is a 19 character ID which should not be allowed. @harrygcollins can you please investigate

Reply:
managed to fix the RegEx